src/components/collapsible.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactNode, FC, useState } from "react"
+import classNames from "classnames"
+
+import ArrowDown from "./icons/arrowDown"
+
+interface Props {
+  /**
+   * False if the element shall be expanded by default
+   * default: true
+   */
+  isInitiallyCollapsed?: boolean
+  /**
+   * A render prop to customize the clickable title of the collapsible element
+   *  - isCollapsed tells you the current state of the collapsible element
+   */
+  renderToggle: (options: { isCollapsed: boolean }) => ReactNode
+  /**
+   * Children to be rendered within the collapse component. The elements will be visible when the component is expanded.
+   */
+  children?: () => ReactNode
+  /**
+   * Function executed when the collapse state has changed.
+   *  - passes the new collapse state as a function parameter
+   */
+  onCollapseStateChange?: (isCollapsed: boolean) => void
+  /**
+   * The size of the arrow down icon
+   * default: 32
+   */
+  collapseIconSize?: number
+  /**
+   * If the element should get a bit transparent on hover
+   * default: true
+   */
+  opacityOnHover?: boolean
+}
+
+const Collapsible: FC<Props> = ({
+  renderToggle,
+  isInitiallyCollapsed = true,
+  children,
+  onCollapseStateChange,
+  collapseIconSize = 32,
```

Comment:
I dislike passing exact sizes in pixels as props. The reason is that it's pretty easy to misuse it and allow all sorts of icon sizes, while the styleguide actually limits them. Especially in a shared component package, I would recommend to be as strict as possible and not allow any unwanted values

How many sizes are you experiencing in the projects? Can we settle for a toggle like `"large" | "small"` or something?

Reply:
to be honest i would argue why is neccesary to have different sizes if the icons. what are the use cases for having different sizes? afaik after discussing with designers is that if we have an icon of 24px we should not make it bigger to like 32px, but we should use a different icon due to the stroke.

Reply:
we only use 24, 32 and 40 (dealerhub listing overview page). So I could go with "small", "medium", and "large".

Reply:
> to be honest i would argue why is neccesary to have different sizes if the icons. what are the use cases for having different sizes? afaik after discussing with designers is that if we have an icon of 24px we should not make it bigger to like 32px, but we should use a different icon due to the stroke.

because dealerhub uses a larger icon on the overview page. That's why I am having this as a prop. 
On listings, the icon is always 32x32 as it is at the moment

Reply:
i don't think we should use 40px at all. i think we should check with the designer the cases we use 24, 32 and 40 and make the uniform. 40 is definitely not a number to use.
and regarding using 24 and 32, as i mentioned before, they suggest using 2 different icons due to the stroke. 

my advice would be to check with Fred or Enrique the cases where we use 32 and 40 and try to see if we can use 24 in both case.

Reply:
fair. I would still bring it to the attention of a designer, for example Enrique and verify their intent was to have three sizes. often they're not aware and very happy to simplify the design system

Reply:
![image](https://user-images.githubusercontent.com/71456764/102471310-5bb63e00-4055-11eb-885a-c1fbf6b9219e.png)

As we currently only use 32px in both projects, I'll drop the prop completely

Reply:
I also imported the new icon he sent me from here: https://www.figma.com/file/L06vFtqz3AhJbbdiGWPuzy/Myicons-v1.16?node-id=40%3A1986 to be completely up to date 👍

Reply:
thanks a lot for aligning this 🙏